prototype_source/pt2e_quant_ptq_static.rst

```diff
@@ -0,0 +1,399 @@
+
+Also there are a few things that can be improved on:
+(3). Currently we use QConfigMapping and BackendConfig as separate objects, QConfigMapping describes user’s intention of how they want their model to be quantized, BackendConfig describes what kind of quantization a backend support, currently BackendConfig is backend specific, but QConfigMapping is not, and user can provide a QConfigMapping that is incompatible with a specific BackendConfig, this is not a great UX. Ideally we can structure this better by making both configuration (QConfigMapping) and quantization capability (BackendConfig) backend specific, so there will be less confusion about incompatibilities.
+
+(4). Currently in QConfig we are exposing observer/fake_quant observer classes as an object for user to configure quantization, this increases the things that user may need to care about, e.g. not only the dtype but also how the observation should happen, these could potentially be hidden from user so that the user interface is simpler.
+
+Here is a summary of benefits with the quantizer API:
+- Programmability (Addressing (1) and (2)): When a user’s quantization needs are not covered by available quantizers, users can build their own quantizer and compose it with other quantizers as mentioned earlier.
+- Simplified UX (Addressing (3)): Provides a single instance with which both backend and users interact. Thus you no longer have 1) user facing quantization config mapping to map users intent and 2) a separate quantization config that backends interact with to configure what backend support. We will still have a method for users to query what is supported in a quantizer. With a single instance, composing different quantization capabilities also becomes more natural than previously. For example QNNPACK does not support embedding_byte and we have native support for this in ExecuTorch. Thus if we had ExecuTorchQuantizer that only quantized embedding_byte, then it can be composed with QNNPACKQuantizer. (Previously this will be concatenating the two BackendConfig together and since options in QConfigMappings are not backend specific, user also need to figure out how to specify the configurations by themselves that matches the quantization capabilities of the combined backend. with a single quantizer instance, we can compose two quantizers and query the composed quantizer for capabilities, which makes it less error prone and cleaner, e.g. composed_quantizer.quantization_capabilities())
```

Comment:
```suggestion
- Simplified and improved User Experience (UX) (addressing 3.): Provides a single instance with which both backend and users interact. Thus, you no longer have 1) user-facing quantization config mapping to map user's intent and 2) a separate quantization config that backends interact with to configure which backend to support. We will still have a method for users to query what is supported in a quantizer. With a single instance, composing different quantization capabilities also becomes more natural than previously. For example, QNNPACK does not support ``embedding_byte`` and we have native support for this in ExecuTorch. Thus, if we had ``ExecuTorchQuantizer`` that only quantized ``embedding_byte``, then it can be composed with ``QNNPACKQuantizer``. (Previously this will be concatenating the two ``BackendConfig`` together and since options in ``QConfigMappings`` are not backend specific, user also need to figure out how to specify the configurations by themselves that matches the quantization capabilities of the combined backend. With a single quantizer instance, we can compose two quantizers and query the composed quantizer for capabilities, which makes it less error prone and cleaner, such as ``composed_quantizer.quantization_capabilities())``.
```